e2e/steps/delete-product-request.ts

```diff
@@ -25,12 +18,14 @@ And(/^I log out$/, function () {
   cy.get('.nav-link').contains('Logout').click();
 });
 
-// This is just to avoid interferences between requests test, and it also helps us to check a error msg
+// This is just to avoid interferences between requests test, and it also helps us to check an error msg
 Then(/^I delete the "([^"]*)" request to avoid interfering other tests$/, (requestName) => {
   cy.get('.nav-link').contains("Requests").click();
   cy.url().should('include', '/requests');
   cy.wait(1000);
   cy.get('.card-text').contains(requestName).click();
+  cy.wait(2000);
+  cy.get('button').contains("Delete").click();
```

Comment:
These 3 clicks to the Delete button seem to be one of the origins of the problems with the tests. Really, there are just two buttons, one to ask for deletion and the other to confirm. Use just 2 clicks with a check in the middle for the presence of the text "confirm deletion"